prov/util/src/util_attr.c

```diff
@@ -1134,6 +1134,11 @@ void ofi_alter_info(struct fi_info *info, const struct fi_info *hints,
 		return;
 
 	for (; info; info = info->next) {
+		if ((info->caps & (FI_RMA | FI_ATOMICS)) &&
+		     OFI_IS_MR_BASIC(info) &&
+		     (!hints || !hints->domain_attr || !OFI_IS_MR_BASIC(hints)))
+		     info->caps &= ~(FI_REMOTE_WRITE | FI_REMOTE_READ);
```

Comment:
I want to avoid re-introducing the OFI_IS_MR_BASIC macro.  It obscures the check that needs to be made.  This also looks like a provider specific restriction, rather than a general one.

Reply:
Why do you think it obscures the check? The alternative is to just expand the macro inline, twice. This change handles any provider that requires FI_MR_BASIC so I think it's generic enough to be in ofi_alter_info.

Reply:
The following MR flags are applicable to being a target of RMA operations: FI_MR_RAW, FI_MR_VIRT_ADDR, FI_MR_PROV_KEY, and FI_MR_RMA_EVENT.  OFI_IS_MR_BASIC only includes half of these, plus FI_MR_ALLOCATED, which is independent from being an RMA target.

The check would be more explicit being in a separate call, so that the name indicates what is happening.  Ideally, there would a check that could be paired with this as part of the hints validation.  Otherwise, we end up with the provider doing the hints validation portion, with common code assuming that, and only adjusting the output.